### ISSUE TITLE
composite outputs weird pixels to the bottom of the image

### ISSUE THREAD

**Issue by NijiharaTsubasa** [oliver-moran/jimp#406]
Bug reproduces on Node.js starting at 4.x up to 9.x on both Windows and CentOS.

Here's the source image I'm using:

(NOTE the image is NSFW, sorry I just can't find another image that reproduces this)

https://card.llsif.moe/asset/assets/image/units/u_normal_navi_22109003.png?gettexb

Here's the code: 

````javascript
const fs = require('fs')
const Jimp = require('jimp')

let img = new Jimp(512,720)
Jimp.read('./u_normal_navi_22109003.png').then(src => {
  img.composite(src.scale(0.9), -180, -142)
  img.getBuffer(Jimp.MIME_PNG, (err, res) => {
    fs.writeFileSync('card.png', res)
  })
})
````

And the output...

https://user-images.githubusercontent.com/7794838/36292189-b3cf49ea-130a-11e8-96be-16de68c3d4d9.png

I think the outputted image itself is self-explainable... Look at the bottom of it...

And if we remove the `.scale(0.9)` in the code it still reproduces

**Comment by czycha** [comment]
I couldn't quite figure out what I was seeing with the image you gave so I experimented with something else.

## Test environment

```js
const fs = require('fs')
const Jimp = require('jimp')

let img = new Jimp(400,400)
Jimp.read('./kat.png').then(src => {
  img.composite(src, 200, -(src.bitmap.height / 2))
  img.write('card.png');
});
```

**kat.png:**
![kat](https://user-images.githubusercontent.com/4816472/36620486-e7052132-18b8-11e8-8856-9cae070e4308.png)

## Running with Jimp 0.2.28

**card.png:**
![card](https://user-images.githubusercontent.com/4816472/36620508-02607a80-18b9-11e8-90dd-3fc39d78ec71.png)

This looks to me like the image is just wrapping to the start of the next line to write the next pixel.

## Running with Jimp@HEAD

**card.png:**
![card](https://user-images.githubusercontent.com/4816472/36621001-7e9a043e-18bb-11e8-9284-69ae139bca34.png)

This is completely different. At the top right edge, it looks like there's a line of pixels that are either basketball or skin colored, although that should be way off-canvas. On the right edge, you can see a line of pixels that matches his shoe, as well as white and navy lines to match his uniform. My guess is that it is writing either the first or last (depending on which edge) opaque pixel that it encounters off-canvas. Perhaps that pixel is saved in memory and erroneously is written on the edge instead of what should actually be there.

## TL;DR

There's some weird issues where the composite is either overflowing the canvas or writing the last opaque pixel in memory on the edge.

**Comment by hipstersmoothie** [comment]
Tried in current version (v0.3.4) with no problems

![anime-test](https://user-images.githubusercontent.com/1192452/43998598-104e2082-9dae-11e8-8c11-f47e6bc02b35.png)
